src/mopidy_spotify/playlists.py

```diff
@@ -48,18 +47,32 @@ def _get_playlist(self, uri, *, as_items=False):
     def refresh(self):
         if not self._backend._web_client.logged_in:
             return
-
-        logger.info("Refreshing Spotify playlists")
-
-        with utils.time_logger("playlists.refresh()", logging.DEBUG):
-            self._backend._web_client.clear_cache()
-            count = 0
-            for playlist_ref in self._get_flattened_playlist_refs():
-                self._get_playlist(playlist_ref.uri)
-                count = count + 1
-            logger.info(f"Refreshed {count} Spotify playlists")
-
-        self._loaded = True
+        if self._refreshing:
+            logger.info("Refreshing Spotify playlists already in progress")
+            return
+        try:
+            uris = [ref.uri for ref in self._get_flattened_playlist_refs(refresh=True)]
+            logger.info(f"Refreshing {len(uris)} Spotify playlists in background")
+            threading.Thread(
+                target=self._refresh_tracks,
+                args=(uris,),
+                daemon=True,
+            ).start()
+        except Exception:
+            logger.exception("Error occurred while refreshing Spotify playlists")
+
+    def _refresh_tracks(self, playlist_uris):
+        self._refreshing = True
+        try:
+            with utils.time_logger("playlists._refresh_tracks()", logging.DEBUG):
+                refreshed = [uri for uri in playlist_uris if self.get_items(uri)]
```

Comment:
We could also just use the length of the uris from the input?

Reply:
I was trying to filter out playlists that didn't refresh successfully with the ` if self.get_items(uri)`. Originally I used `_get_playlist` here instead but for some reason changed it to `get_items`, in retrospect that was a bad change since what I have now will incorrectly also filter out playlists with no tracks.